nsiqcppstyle_exe.py

```diff
@@ -371,40 +409,22 @@ def __init__(self, fileFilterPath, extLangMap, varMap, activeFilterName):
         filter = self.GetFilter(self.defaultFilterName)
         self.activeFilterName = self.defaultFilterName
 
+        if filterString:
+            lines = filterString.split(';')
```

Comment:
This looks fragile to me. Is there a chance the user could use `;` for some other reason except as a delimiter?

---
If there's a chance...
* Maybe we can use a better inline format such as json/yaml/csv which allows for escaping the literal.
* Or the user could use multi line string as well, but that's a bit complicated in most command lines

Reply:
I selected semicolon (';') because it supposedly should not occur in NsiqCppStyle filter lines.  Having said that, a filter line can contain a file system path, and under Linux practically any character including a semicolon (excepting NUL and '/'), is a valid file/folder name character.

In addition, a filter line can start with '#', denoting a comment that should be ignore.  This comment can contain a semicolon, which could complicate things.

- You mention using some other inline format, e.g., JSON/YAML/CSV.  Would this complicate things for, say, a non-YAML programmer?  Also, can the Windows CMD parser handle the escaping?  If possible, I'd like to keep the command line string format as standard as possible.
- I agree with you that creating a multi-line command line string can complicate things, both under Windows and under Linux.

I would like to add other possible solutions to the problem of passing a multi-value, delimited string on the command line:

- `--filter-string=` should accept only **one** filter line string, but allow multiple occurrences of this option.  This obviates the need for a delimiter character.  Thus, the command line might look like this:

```
python nsiqcppstyle.py --filter-string="~ TOOL_trace_nsiqcppstyle_callbacks"  --filter-string="~ RULE_4_2_A_A_space_around_operator" test.cpp
```

- The first character in the string passed via `--filter-string=` should denote the delimiter character.  For example, to declare that the delimiter is the vertical bar ('|'), we would use:

```
python nsiqcppstyle.py --filter-string="|~ TOOL_trace_nsiqcppstyle_callbacks|~ RULE_4_2_A_A_space_around_operator" test.cpp
```

- A variation on the previous bullet is to allow specifying a multi-character string as the delimiter.  We could allow the delimiter string to be any number of characters, followed by a space, followed by the filter strings.  So, as an example:

```
python nsiqcppstyle.py --filter-string="/// ~ TOOL_trace_nsiqcppstyle_callbacks///~ RULE_4_2_A_A_space_around_operator" test.cpp
```

Reply:
We are not pioneers of this, and other interfaces use:
* repeated arguments
* json input

Since our format doesn't match json at all, lets go with repeated arguments given the complexity of multi-line input as well as heredoc style delimits

Reply:
Done.